src/components/icons/Sort.tsx

```diff
@@ -1,15 +1,36 @@
 import { useSelector } from 'react-redux'
 import IconType from '../../@types/IconType'
 import State from '../../@types/State'
+import ThoughtId from '../../@types/ThoughtId'
 import { HOME_PATH } from '../../constants'
 import getSortPreference from '../../selectors/getSortPreference'
+import getThoughtById from '../../selectors/getThoughtById'
 import head from '../../util/head'
 import AnimatedIcon from './AnimatedIcon'
 import animationData from './animations/10-sort_4.json'
 import animationDataDesc from './animations/11-sort-descanding_7.json'
 
+/** Find the parent ID of a given thought ID. */
+const findParentId = (state: State, thoughtId: ThoughtId): ThoughtId | null => {
+  const thought = getThoughtById(state, thoughtId)
+  return thought ? thought.parentId : null
```

Comment:
This is fine as-is, but a small suggestion:
```suggestion
  return thought?.parentId ?? null
```